pkg/scheduler/core/equivalence_cache.go

```diff
@@ -71,9 +71,9 @@ func (ec *EquivalenceCache) RunPredicate(
 	equivClassInfo *equivalenceClassInfo,
 	cache schedulercache.Cache,
 ) (bool, []algorithm.PredicateFailureReason, error) {
-	ec.mu.Lock()
-	defer ec.mu.Unlock()
+	ec.mu.RLock()
 	fit, reasons, invalid := ec.lookupResult(pod.GetName(), nodeInfo.Node().GetName(), predicateKey, equivClassInfo.hash)
+	ec.mu.RUnlock()
```

Comment:
Please move RLock()/RUnlock() inside `lookupResult ` and `defer ec.mu.RUnlock()`

Reply:
thx, i will fix it.